src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/SPARQL2GraphQLTranslatorImpl.java

```diff
@@ -0,0 +1,454 @@
+            final Set<String> allScalarURI = URI2GraphQLHelper.getPropertyURIs(sgpType,GraphQLFieldType.SCALAR,config,endpoint);
+
+            for(final String uri : allObjectURI){
+                addEmptyObjectField(fieldPaths,subgraphPatterns,connectors,config,endpoint,
+                    currentPath,sgpType,uri);
+            }
+
+            for(final String uri : allScalarURI){
+                addScalarField(fieldPaths,config,currentPath,uri);
+            }
+
+        }
+        else{
+            // If no variable predicate exist, only the necessary fields from the SGP have to be added.
+            for(final TP t : subgraphPatterns.get(subgraphNode)){
+                final Node predicate = t.getTriplePattern().asJenaTriple().getPredicate();
+                if(predicate.isURI() && URI2GraphQLHelper.containsPropertyURI(predicate.getURI(),config,endpoint)){
+                    final String withoutPrefix = config.removePropertyPrefix(predicate.getURI());
+                    final String propertyName = config.removePropertySuffix(withoutPrefix);
+                    if(endpoint.getGraphQLFieldType(sgpType,propertyName) == GraphQLFieldType.OBJECT){
+                        addEmptyObjectField(fieldPaths,subgraphPatterns,connectors,config,endpoint,
+                            currentPath,sgpType,predicate.getURI());
+                    }
+                    else{
+                        addScalarField(fieldPaths,config,currentPath,predicate.getURI());
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Helper function to add a field to the query that represents a nested object. Fields in that nested 
+     * object are added recursively through addSgp
+     */
+    protected void addObjectField(final TreeSet<String> fieldPaths, final Map<Node,Set<TP>> subgraphPatterns, 
+            final Map<Integer,Node> connectors, final GraphQL2RDFConfiguration config, final GraphQLEndpoint endpoint, 
+            final Node nestedSubgraphNode, final String currentPath, final String sgpType, final String predicateURI){
+
+        final String alias = config.removePropertyPrefix(predicateURI);
+        final String fieldName = config.removePropertySuffix(alias);
+        final StringBuilder field = new StringBuilder("object_").append(alias).append(":").append(fieldName);
+        final StringBuilder newPath = new StringBuilder(currentPath).append(field).append("/");
+        final String nestedType = endpoint.getGraphQLFieldValueType(sgpType, fieldName);
+
+        addSgp(fieldPaths,subgraphPatterns,connectors,config,endpoint,
+            nestedSubgraphNode,newPath.toString(),nestedType);
+    }
+
+    /**
+     * Helper function to add a field to the query that represents a nested object that only needs an id field (no more recursive calls)
+     */
+    protected void addEmptyObjectField(final TreeSet<String> fieldPaths, final Map<Node,Set<TP>> subgraphPatterns, 
```

Comment:
You don't use `subgraphPatterns` nor `connectors`, please remove them from the argument list